packages/rabbitmq/src/utils.ts

```diff
@@ -0,0 +1,128 @@
+import { CSL, Cardano, cslToCore } from '@cardano-sdk/core';
+import { OutsideOfValidityInterval } from '@cardano-ogmios/schema';
+import { fromSerializableObject, toSerializableObject } from '@cardano-sdk/util';
+
+export interface DeserializedError {
+  innerError?: {
+    name: string;
+    message: string;
+  };
+  name: string;
+  message: string;
+}
+
+export interface SerializedError {
+  value: {
+    innerError?: {
+      name: string;
+      message: string;
+      stack?: string;
+    };
+    name: string;
+    message: string;
+    stack?: string;
+  };
+}
+
+export const TX_SUBMISSION_QUEUE = 'cardano-tx-submit';
+
+/**
+ * Enriches an error with the right prototype
+ *
+ * @param err the error to enrich
+ */
+const addPrototypeToError = (err: DeserializedError) => {
+  const { name, innerError } = err;
+  let prototype: object | null = Error.prototype;
+
+  if (name in Cardano.TxSubmissionErrors)
+    prototype = Cardano.TxSubmissionErrors[name as keyof typeof Cardano.TxSubmissionErrors].prototype;
+
+  Object.setPrototypeOf(err, prototype);
+
+  if (innerError) addPrototypeToError(innerError);
+};
+
+/**
+ * Deserializes a serialized error
+ *
+ * @param serializedError the error to deserialize
+ */
+export const deserializeError = (serializedError: SerializedError): unknown => {
+  const ret = fromSerializableObject(serializedError) as DeserializedError;
```

Comment:
Also, you can use the existing `getErrorPrototype` option instead of setting it manually:
```suggestion
  const getErrorPrototype = (err: unknown) => {
    if (typeof error === 'object' && typeof error?.name === 'string' && typeof error?.message === 'string') {
      const rawError = error as Cardano.TxSubmissionError;
      const txSubmissionErrorName = rawError.name as keyof typeof Cardano.TxSubmissionErrors;
      const ErrorClass = Cardano.TxSubmissionErrors[txSubmissionErrorName];
      if (ErrorClass) return ErrorClass.prototype;
    }
    return Error.prototype;
  }
  return fromSerializableObject(serializedError, {getErrorPrototype})
```

Reply:
I tried but I wasn't able to use it, now I think I understood how the option works; I can try again to use it.
Thank you for the example.

Reply:
A part from some minor changes since `error?.name` do not compile since `error: unknown`, it works, thank you